### ISSUE TITLE
Webhooks cannot direct message the user who created the webhook

### ISSUE THREAD

**Issue by Roy-Orbison** [mattermost/mattermost#16113]
#### Summary
Exactly as mentioned in the [known Slack compatibility issues][0], an incoming webhook that a user creates cannot direct message that user.

#### Steps to reproduce
Create a webhook and try to limit it to direct messages.

#### Expected behavior
Be able to select DMs from _Channel_ dropdown.

#### Observed behavior (that appears unintentional) 
Can only select actual, named channels.

---
It would be highly useful for a hook to be used for user-specific interaction. I am able to create a direct message conversation with myself, labelled `roy (you)`, so surely a hook posting there is preferable to not at all.

[0]: https://docs.mattermost.com/developer/webhooks-incoming.html#known-slack-compatibility-issues

**Comment by amyblais** [comment]
@levb / @mickmister Do we have a ticket for this or should I create one?

**Comment by mickmister** [comment]
@amyblais I'm not aware of a Jira ticket for this, but I think it makes sense to discuss the UX here with @Roy-Orbison.

Automated DMs usually come from a bot user in the DM channel between the user and the bot user, so putting these webhook messages into a self-DM channel seems a bit off to me. The incoming webhooks allow you to specify the `username` that is used to create the post, which could be used to determine which DM channel to use, but it seems a bit error-prone. @aaronrothschild Do you have any thoughts on this?

**Comment by Roy-Orbison** [comment]
I agree that it's a bit off using a self-direct-message channel, I surmised that the inability to message oneself via a hook was a technical deficiency, and using that method would work around it, at least until a better solution is available.

I have an extra user account on our server that is used only for integrations, a pseudo-bot account named as such. If I send a DM to myself through an incoming webhook owned by that account, it appears under the direct channel of the pseudo-bot. To me, this is no better or worse than a DM from myself if I own the webhook.

**Comment by mickmister** [comment]
I've created a ticket for discussion with the team https://mattermost.atlassian.net/browse/MM-30291

**Comment by Roy-Orbison** [comment]
Seems I should have tested this more; a hook **can** DM the hook owner, but a self-DM cannot be set as the default Channel, so _Lock to this channel_ has to be unchecked. The hook can therefore DM anyone with the `{"channel":"@anyuser","text":"…"}` syntax.

**Comment by mickmister** [comment]
@Roy-Orbison Do you mind if I close this ticket, as it seems the use case is already supported? I would also think that this is a better method of accomplishing this, as it is more flexible and doesn't require a webhook per user.

**Comment by Roy-Orbison** [comment]
@mickmister But it isn't, really. My understanding is that the purpose of restricting a webhook is prevention of abuse. One can give a webhook address to a third party and know where their messages can be sent and how each is labelled. If a webhook must be unrestricted in order to DM its owner, it can DM anyone at all and the message can appear to originate from the hook owner. Even an unintentionally misbehaving service could end up harassing other users, via such a hook. Being able to limit messages via a hook to oneself makes a lot of sense to me.

**Comment by lieut-data** [comment]
Thanks for reporting this issue and for the community discussion around it. After multiple years of inactivity, we're closing this to help keep our issue tracker focused on active work. If this is still relevant, please feel free to open a new issue with updated context and we'll be happy to take another look.
